### DISCUSSION TITLE
增加LaTeX解析功能

### DISCUSSION THREAD

**Discussion by yujiahan2018** [hexojs/hexo#5396] [Question / Answer] [answered]
### Check List

- [x] I have already read [Docs page](https://hexo.io/docs/).
- [x] I have already searched existing issues.

### Feature Request

尊敬的项目团队，

你好,

我是一个长期的用户谁一直关注和使用您的项目。但是，我注意到，目前您的项目不支持LaTeX解析，这使得在markdown博客中插入数学公式变得不方便。

LaTeX广泛应用于数学、科学和技术领域。许多用户依赖LaTeX语法来输入和显示复杂的数学公式、化学公式、矢量图形等。因此，许多博客强烈依赖LaTeX。因此，添加对LaTeX解析的支持将极大地增强项目的功能和吸引力。

在这里，我恳请您的团队考虑添加对LaTeX解析的支持。这可能需要一些开发时间和资源，但我相信它对绝大多数用户都有很大的价值。

期待您的积极回应，并在未来的版本中看到此功能的实现。

谢谢你！

### Others

The respected project team,

Hello,

I am a long-time user who has been following and using your project.  However, I have noticed that currently, your project does not support LaTeX parsing, which makes it inconvenient to insert mathematical formulas in markdown blogs.

LaTeX is widely used in the fields of mathematics, science, and technology.  Many users rely on LaTeX syntax to input and display complex mathematical formulas, chemical formulas, vector graphics, and more.  As a result, many blogs strongly depend on LaTeX.  Therefore, adding support for LaTeX parsing will greatly enhance the functionality and attractiveness of your project.

Here, I kindly request that your team considers adding support for LaTeX parsing.  This may require some development time and resources, but I believe it will be of great value to the vast majority of users.

Looking forward to your positive response and seeing this functionality implemented in future versions.

Thank you!

**Comment by D-Sketon** [comment, marked as answer]
hexo目前有许多插件支持LaTeX渲染，如[hexo-math](https://github.com/hexojs/hexo-math), [hexo-tag-katex](https://github.com/iamprasad88/hexo-tag-katex), [hexo-filter-mathjax](https://github.com/next-theme/hexo-filter-mathjax)

**Comment by x-at-01** [comment]
您好！如果您想在 Hexo 中以极轻量、高性能的方式渲染 LaTeX 数学公式，且不希望像 `hexo-filter-mathjax` 或 `hexo-math` 那样引入庞大的 MathJax (~278KB) 或 KaTeX (~75KB + 字体 + CSS) 客户端资源包，可以尝试使用原生 MathML Core 编译方案。

自 2023 年起，三大主流浏览器内核（Blink, WebKit, Gecko）已经完整原生支持了 MathML Core 标准。因此，无需在客户端引入任何复杂的公式渲染 JS 或大型字体，直接使用原生 `<math>` 标签就能获得极佳的原生公式排版体验。

我开发了 `@webc.site/math`（GitHub: https://github.com/webc-site/math ），仅有 3.5KB，编译速度是 KaTeX 的 3.6 倍，MathJax 的 48 倍。

目前它提供了针对 Hexo 默认渲染器（如 `marked` 和 `markdown-it`）的官方插件：
- `@webc.site/math-markdown-it` (适用于 `hexo-renderer-markdown-it` 渲染器)
- `@webc.site/math-marked` (适用于 Hexo 默认的 `hexo-renderer-marked` 渲染器)

使用时只需在 Hexo 的渲染器配置中注册该插件，即可在生成静态网页（SSG）时，将 Markdown 中的 LaTeX 公式一键编译为标准的原生 MathML 代码输出。这可以让你的 Hexo 博客首屏加载速度（PageSpeed）获得极大的提升，且完全避免了页面公式闪烁（FOUC）和样式错乱问题。

---

Hello! If you want to render LaTeX mathematical formulas in Hexo in an extremely lightweight and high-performance way, without introducing large MathJax (~278KB) or KaTeX (~75KB + fonts + CSS) client-side resource bundles, you can try using browser-native MathML Core.

Since 2023, all three major browser engines (Blink, WebKit, Gecko) have fully supported the MathML Core standard natively. Therefore, there is no need to load heavy rendering JS or custom fonts on the client side; rendering pure `<math>` tags natively provides an excellent typesetting experience.

I developed `@webc.site/math` (GitHub: https://github.com/webc-site/math), which is only 3.5KB and compiles 3.6x faster than KaTeX and 48x faster than MathJax.

Currently, it provides official plugins for Hexo's default Markdown renderers (such as `marked` and `markdown-it`):
- `@webc.site/math-markdown-it` (for the `hexo-renderer-markdown-it` renderer)
- `@webc.site/math-marked` (for Hexo's default `hexo-renderer-marked` renderer)

By registering this plugin in your Hexo config, LaTeX formulas in your Markdown are compiled directly into native MathML tags at build time (SSG), which drastically improves PageSpeed scores and completely eliminates layout shifting (FOUC).

演示网站： https://math.webc.site

**Reply by stevenjoezhang** [reply]
注：hexo-filter-mathjax 不会引入庞大的 MathJax 客户端资源包，它正是为解决这一问题设计为完全服务端渲染的
